async.c

```diff
@@ -317,22 +321,20 @@ static void __redisAsyncFree(redisAsyncContext *ac) {
     while (__redisShiftCallback(&ac->sub.invalid,&cb) == REDIS_OK)
         __redisRunCallback(ac,&cb,NULL);
 
-    /* Run subscription callbacks with NULL reply */
-    if (ac->sub.channels) {
-        dictInitIterator(&it,ac->sub.channels);
-        while ((de = dictNext(&it)) != NULL)
-            __redisRunCallback(ac,dictGetEntryVal(de),NULL);
-
-        dictRelease(ac->sub.channels);
-    }
+    #define RUN_SUB_CALLBACKS(cb_dict) \
```

Comment:
Wanted to have the logic for iteration matching scenarios for future update cases